suricata/update/main.py

```diff
@@ -424,26 +424,32 @@ def manage_classification(suriconf, files):
 def handle_dataset_files(rule, dep_files):
     if not rule.enabled:
         return
-    load_attr = [el.strip() for el in rule.dataset.split(",") if el.startswith("load")]
-    state_attr = [el.strip() for el in rule.dataset.split(",") if el.startswith("state")]
-    if not load_attr and not state_attr:
-        return
-    if load_attr and state_attr:
-        logger.error("Invalid dataset rule")
+
+    dataset_load = [el.strip() for el in rule.dataset.split(",") if el.startswith("load")]
+    if not dataset_load:
+        # No dataset load found.
         return
-    elif load_attr:
-        ds_attr = load_attr[0]
-    elif state_attr:
-        ds_attr = state_attr[0]
+    dataset_filename = dataset_load[0].split(maxsplit=1)[1].strip()
```

Comment:
I don't think it makes sense to handle state, or for state to be used in distributed rulesets.

If state is used, state will get stored.  Then on the next rule update it will be wiped with ruleset provided state file, and the state has to build up again.

Am I missing a use case here?

Reply:
State is not meant for ruleset distribution of a dataset, but a ruleset rule might still use it. E.g. to track something unique (UA's for example).

Reply:
OK. Thank you. I had thought that since `state` means both `load` and `save`, it could be used to distribute rulesets where the provider may want to give a set of defaults for users (to load) and allow them to update that list (to save).